pkg/tenant/tenants.go

```diff
@@ -67,54 +67,69 @@ func (s *Server) TenantList(c *gin.Context) {
 // Route: /tenant
 func (s *Server) TenantCreate(c *gin.Context) {
 	var (
-		err error
-		t   *api.Tenant
+		err    error
+		t      *tokens.Claims
+		tenant *api.Tenant
 	)
 
-	// TODO: Add authentication and authorization middleware
+	// Fetch tenant from the context.
+	if t, err = middleware.GetClaims(c); err != nil {
+		log.Error().Err(err).Msg("could not fetch tenant from context")
+		c.JSON(http.StatusUnauthorized, api.ErrorResponse(err))
+		return
+	}
+
+	// Get the tenant's organization ID and return a 500 response if it is not a ULID.
+	var orgID ulid.ULID
+	if orgID, err = ulid.Parse(t.OrgID); err != nil {
+		log.Error().Err(err).Msg("could not parse org id")
```

Comment:
One helpful thing for future debugging might be to add a string field to the log call with the orgid we tried to parse. That way it shows up in the logs if it was a badly formatted ULID.

e.g. `log.Error().Err(err).Str("orgid", t.OrgID).Msg("could not parse org id")`